spec/controllers/batch_controller_spec.rb

```diff
@@ -22,6 +22,23 @@
       end
     end
 
+    describe "when submiting files on behalf of another user" do
+      let(:somebody_else_file) do 
+        f = GenericFile.create(title: ['Original Title'])
+        f.apply_depositor_metadata('current_user')
+        f.save!
+        f.on_behalf_of = 'somebody@else.org'
```

Comment:
I think you were getting that error because it couldn't find the user with the email: `somebody@else.org`